File: JavaScript/Sorting/SelectionSort.js
let selectionSort = (arr) => {

   //Loop till the second last element
   for(let i = 0; i < arr.length - 1; i++){

      //Loop after the i till the last element
      for(let j = i + 1; j < arr.length; j++){

         //if jth element is less than the ith element then swap
         //change < to > for sorting in descending order
         if(arr[j] < arr[i]){
           let temp = arr[i];
           arr[i] = arr[j];
           arr[j] = temp;
         }

      }

   }
   
   //return the sorted array
   return arr;
}

File: data structure/selectionsort.cpp
#include <iostream>
using namespace std;
int main()
{

    int arr[]={1,2,3,4,5,6,7};
    int n=7; 
   
   for (int i = 0; i < n-1 ; i++)
   {
    
           int min=i;
      for (int j = i+1; j < n; j++)
   {
     
      if(arr[min]>=arr[j]){
        int temp=arr[min];
        arr[min]=arr[j];
        arr[j]=temp;
       
      }
   }


   }


    for (int i = 0; i < n; i++)
    {
        cout<<arr[i]<<" ";
    }


return 0;
}
